### PR TITLE
Support mocking upload_file()

### PR DESCRIPTION
**Pull request by jonkeane**

### CODE REVIEW

File: R/trace.R [hunk 1]
```diff
@@ -18,6 +18,12 @@ with_trace <- function (x, where=topenv(parent.frame()), print=getOption("httpte
 
 mock_perform <- function (mocker, ...) {
     tracer <- substitute_q(fetch_tracer, list(.mocker=mocker))
+    # trace curl's form_file making the path normalization a no-op so that file
+    # hashes are the same on different platforms
+    trace(curl::form_file, quote(
+        normalizePath <- function (path, ...) { return(path) }
```
Comment (nealrichardson) [inline R/trace.R]: Don't need the `{` `}`

File: R/trace.R [hunk 2]
```diff
@@ -48,6 +54,7 @@ quietly <- function (expr) {
 #' @return Nothing; called for its side effects
 #' @export
 stop_mocking <- function () {
+    quietly(untrace(curl::form_file))
```
Comment (nealrichardson) [inline R/trace.R]: Is there a reason not to use safe_untrace here?

Comment (jonkeane) [reply R/trace.R]: Nope, I've replaced quietly with `safe_untrace`

**Comment by jonkeane** [comment]
Fixed the few notes, and added to NEWS

**Pull request merged**